## `nixpkgs-review` result

Generated using [`nixpkgs-review-gha`](https://github.com/Defelo/nixpkgs-review-gha)

Command: `nixpkgs-review pr 414766`

Logs: https://github.com/guylamar2006/nixpkgs-review-gha/actions/runs/15509253621




---
### `x86_64-darwin` (sandbox = true)
<details>
  <summary>:x: 1 package failed to build:</summary>
  <ul>
    <li>pngpaste</li>
  </ul>
</details>

---
<details>
<summary>Error logs: `x86_64-darwin`</summary>
<details>
<summary>pngpaste</summary>
<pre>Running phase: configurePhase
@nix { "action": "setPhase", "phase": "configurePhase" }
no configure script, doing nothing
Running phase: buildPhase
@nix { "action": "setPhase", "phase": "buildPhase" }
build flags: SHELL=/nix/store/1ys14sy4jzaclkfrwvnqvid0d8p0lawh-bash-5.2p37/bin/bash
clang -Wall -g -O3 -ObjC \
	-framework Foundation -framework AppKit \
	-o pngpaste \
	pngpaste.m
Running phase: installPhase
@nix { "action": "setPhase", "phase": "installPhase" }
install flags: SHELL=/nix/store/1ys14sy4jzaclkfrwvnqvid0d8p0lawh-bash-5.2p37/bin/bash PREFIX=/nix/store/2nys9yksw5mv9lnxyl0i8fi37prwlahh-pngpaste-0.2.3 install
clang -Wall -g -O3 -ObjC \
	-framework Foundation -framework AppKit \
	-o pngpaste \
	pngpaste.m
cp pngpaste /usr/local/bin/
cp: cannot stat '/usr/local/bin/': Operation not permitted
make: *** [Makefile:8: install] Error 1</pre>
</details>
</details>


---
### `aarch64-darwin` (sandbox = true)
<details>
  <summary>:x: 1 package failed to build:</summary>
  <ul>
    <li>pngpaste</li>
  </ul>
</details>

---
<details>
<summary>Error logs: `aarch64-darwin`</summary>
<details>
<summary>pngpaste</summary>
<pre>Running phase: configurePhase
@nix { "action": "setPhase", "phase": "configurePhase" }
no configure script, doing nothing
Running phase: buildPhase
@nix { "action": "setPhase", "phase": "buildPhase" }
build flags: SHELL=/nix/store/8ivrpmp3arvxbr6imdwm2d28q9cjsqvi-bash-5.2p37/bin/bash
clang -Wall -g -O3 -ObjC \
	-framework Foundation -framework AppKit \
	-o pngpaste \
	pngpaste.m
Running phase: installPhase
@nix { "action": "setPhase", "phase": "installPhase" }
install flags: SHELL=/nix/store/8ivrpmp3arvxbr6imdwm2d28q9cjsqvi-bash-5.2p37/bin/bash PREFIX=/nix/store/6z09m9l2bakihgananaxqh4zg9bv74hi-pngpaste-0.2.3 install
clang -Wall -g -O3 -ObjC \
	-framework Foundation -framework AppKit \
	-o pngpaste \
	pngpaste.m
cp pngpaste /usr/local/bin/
cp: cannot stat '/usr/local/bin/': Operation not permitted
make: *** [Makefile:8: install] Error 1</pre>
</details>
</details>

--- a/pkgs/by-name/pn/pngpaste/package.nix
+++ b/pkgs/by-name/pn/pngpaste/package.nix
@@ -4,22 +4,21 @@
   fetchFromGitHub,
 }:
 
-let
+stdenv.mkDerivation (finalAttrs: {
   pname = "pngpaste";
   version = "0.2.3";
-in
-stdenv.mkDerivation {
-  inherit pname version;
+
   src = fetchFromGitHub {
     owner = "jcsalterego";
-    repo = pname;
-    rev = version;
+    repo = "pngpaste";
+    tag = finalAttrs.version;
     sha256 = "uvajxSelk1Wfd5is5kmT2fzDShlufBgC0PDCeabEOSE=";
   };
 
   installPhase = ''
-    mkdir -p $out/bin
-    cp pngpaste $out/bin
+    runHook preInstall
+    install -Dm555 pngpaste $out/bin
+    runHook postInstall
   '';
 
   meta = {
@@ -32,9 +31,9 @@ stdenv.mkDerivation {
       falling back to PNG.
     '';
     homepage = "https://github.com/jcsalterego/pngpaste";
-    changelog = "https://github.com/jcsalterego/pngpaste/raw/${version}/CHANGELOG.md";
+    changelog = "https://github.com/jcsalterego/pngpaste/raw/${finalAttrs.version}/CHANGELOG.md";
     platforms = lib.platforms.darwin;
     license = lib.licenses.bsd2;
     maintainers = with lib.maintainers; [ samw ];
   };
-}
+})

